tests/bluetooth/host/mock_host/src/net.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include <stddef.h>
+#include <string.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <zephyr/net/buf.h>
+#include <zephyr/bluetooth/buf.h>
+
+
+static uint8_t *fixed_data_alloc(struct net_buf *buf, size_t *size,
+			      k_timeout_t timeout)
+{
+	ztest_test_fail();
+	return NULL;
+}
+
+static void fixed_data_unref(struct net_buf *buf, uint8_t *data)
+{
+	/* Nothing needed for fixed-size data pools */
+	ztest_test_fail();
+}
+
+const struct net_buf_data_cb net_buf_fixed_cb = {
+	.alloc = fixed_data_alloc,
+	.unref = fixed_data_unref,
+};
+
+struct net_buf *net_buf_alloc_fixed(struct net_buf_pool *pool, k_timeout_t timeout)
+{
+	zassert_not_null(pool, "Value was NULL");
+	ztest_check_expected_value(pool);
+	hooks_net_buf_alloc_fixed_timeout_validation_hook(timeout.ticks);
+	return (struct net_buf *)ztest_get_return_value_ptr();
+}
+
+void net_buf_simple_reserve(struct net_buf_simple *buf, size_t reserve)
+{
+	ztest_check_expected_value(buf);
+	zassume_equal(reserve, BT_BUF_RESERVE, "Invalid value");
```

Comment:
why is that assumption here ? I think this should rather be a `ztest_check_expected_value(reserve)`. If it's a default value you want to check for, you should use an assert with `BT_BUF_RESERVE`. 

But I think you should setup the mocking properly for that param too in the test cases instead of hardcoding an assert.

The need for suite-based defaults might be solved by FFF (haven't read the docs yet). I know CMocka has a semi-documented feature for that (using negative value as counts, see [will_return_count](https://api.cmocka.org/group__cmocka__mock.html#gaac9a382a8509489d619538fe264e66a7), so maybe FFF also does.

When that is supported you can just have a suite setup fn that sets up those default mocks and then not worry about it in the individual testcases, while not having to hardcode values in the mocks themselves.